ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_manipulation.py

```diff
@@ -1015,3 +1015,49 @@ def test_expand(
         x=x[0],
         shape=shape,
     )
+
+
+@st.composite
+def _concat_from_sequence_helper(draw):
+    shape = draw(st.shared(helpers.get_shape(min_num_dims=1), key="values_shape"))
+    num_arrays = draw(
+        st.shared(helpers.ints(min_value=1, max_value=3), key="num_arrays")
+    )
+    dtype = draw(st.sampled_from(draw(helpers.get_dtypes("valid"))))
+    dtypes = [dtype for _ in range(num_arrays)]
+    arrays, _ = helpers.dtype_and_values(
+        dtype=dtype, shape=shape, num=num_arrays, ret_shape=True
+    )
+    axis = draw(st.shared(helpers.get_axis(shape), key="axis"))
+    return dtypes, arrays, axis
```

Comment:
You don't need to use `st.shared` anywhere. `st.shared` is useful only when we want to access the same exact number that was generated in the previous `draw` that had the same `key`.